cpp/models/abm/README.md

```diff
@@ -1,57 +1,147 @@
-    mio::abm::Location& home = world.add_location(mio::abm::LocationType::Home);
-    mio::abm::Location& school = world.add_location(mio::abm::LocationType::School);
-    mio::abm::Location& work = world.add_location(mio::abm::LocationType::Work);
-    mio::abm::Person& p1 = world.add_person(home, mio::abm::InfectionState::Susceptible);
-    mio::abm::Person& p2 = world.add_person(home, mio::abm::InfectionState::Susceptible);
-    mio::abm::Person& p3 = world.add_person(home, mio::abm::InfectionState::InfectedNoSymptoms);
-    mio::abm::Person& p4 = world.add_person(home, mio::abm::InfectionState::Susceptible);
+```cpp
+auto testing_min_time = mio::abm::days(1);
+auto probability      = 0.5;
+auto start_date       = mio::abm::TimePoint(0);
+auto end_date         = mio::abm::TimePoint(0) + mio::abm::days(30);
+auto test_type        = mio::abm::AntigenTest();
+auto test_at_work     = std::vector<mio::abm::LocationType>{mio::abm::LocationType::Work};
+auto testing_criteria_work =
+    std::vector<mio::abm::TestingCriteria>{mio::abm::TestingCriteria({}, test_at_work, {})};
+auto testing_scheme_work =
+    mio::abm::TestingScheme(testing_criteria_work, testing_min_time, start_date, end_date, test_type, probability);
+world.get_testing_strategy().add_testing_scheme(testing_scheme_work);
+```
+
+For some infections to happen during the simulation, we have to initialize people with infections.
 
-    //setup the simulation
-    double t0   = 0;
-    double tmax = 100;
-    mio::abm::Simulation sim  = mio::abm::Simulation(t0, std::move(world));
+```cpp
+person.add_new_infection(mio::abm::Infection(rng, mio::abm::VirusVariant::Wildtype, person.get_age(), world.parameters, start_date, infection_state));
+```
 
-    sim.advance(tmax);
+We can add restrictions for people after a specific date. For example, only 10% of the people go to social events after day 10.
 
-    //handle result
-    mio::TimeSeries<double>& result = sim.get_result();
-    //...
+```cpp
+auto t_lockdown = mio::abm::TimePoint(0) + mio::abm::days(10);
+mio::abm::close_social_events(t_lockdown, 0.9, world.parameters);
 ```
+
+Then we run the simulation.
+
+```cpp
+sim.advance(mio::abm::TimePoint(0) + mio::abm::days(30));
+```
+
+If we want to track things in the simulation, we need to set up a logger, for example, to track all the TimePoints of each simulation step:
+
+```cpp
+struct LogTimePoint : mio::LogAlways {
+        using Type = double;
+        static Type log(const mio::abm::Simulation& sim)
+        {
+            return sim.get_time().hours();
+        }
+    };
+```
+
+Now we just need to set up the History object and run the simulation with it. As a writer, we use the predefined DataWriterToMemory, which just saves the Timepoints in a vector. Afterwards, one can access it through get_log.
+
+```cpp
+    mio::History<mio::DataWriterToMemory, LogTimePoint> history;
+    sim.advance(tmax, history);
+    auto log = history.get_log();
```

Comment:
This is not a vector of TimePoints as you imply above! Please fix either this code snipped or change the description.

Reply:
done